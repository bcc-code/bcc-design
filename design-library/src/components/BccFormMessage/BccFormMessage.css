@layer components {
    .bcc-form-message, .bcc-input-message {
        @apply text-caption;
        /* default state */
        @apply text-secondary;
    }

    /* State */
    .bcc-form-message-error, .bcc-input-message-error {
        @apply text-danger;
    }
    .bcc-form-message-success, .bcc-input-message-success {
        @apply text-success;
    }
}
