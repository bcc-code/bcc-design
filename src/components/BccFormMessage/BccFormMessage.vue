<script setup lang="ts">
type Props = {
  state?: "default" | "error" | "success";
};

withDefaults(defineProps<Props>(), {
  state: "default",
});
</script>

<template>
  <span
    class="bcc-form-message"
    :class="{
      'bcc-form-message-default': state === 'default',
      'bcc-form-message-error': state === 'error',
      'bcc-form-message-success': state === 'success',
    }"
  >
    <slot />
  </span>
</template>
