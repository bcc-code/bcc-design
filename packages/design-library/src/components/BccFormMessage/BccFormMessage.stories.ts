import BccFormMessage from "./BccFormMessage.vue";

import type { Meta, StoryFn } from "@storybook/vue3";

/**
 * A companion component used with for example the input to show a validation message
 */
export default {
  title: "Forms/BccFormMessage",
  component: BccFormMessage,
  argTypes: {
    state: {
      description: "Style of the message",
      options: ["default", "error", "success"],
      control: { type: "radio" },
    },
  },
} as Meta<typeof BccFormMessage>;

/**
 * Pass content to the default slot to render a message
 */
const Template: StoryFn<typeof BccFormMessage> = (args) => ({
  components: { BccFormMessage },
  setup() {
    return { args };
  },
  template: `
    <BccFormMessage v-bind="args">
      <span v-if="args.slotDefault">
        {{ args.slotDefault }}
      </span>
      <span v-else>
        <strong>Well done!</strong> This username is valid
      </span>
    </BccFormMessage>
  `,
});

export const Example = Template.bind({});
Example.args = {
  slotDefault: "",
  state: "success",
};
Example.parameters = {
  docs: {
    source: {
      language: "html",
      code: `
<BccFormMessage state="success">
  <strong>Well done!</strong> This username is valid
</BccFormMessage>
`,
    },
  },
};

/**
 * Set the `state` prop to control how the input is rendered. Set the `disabled` prop to disable the input
 */
export const State: StoryFn<typeof BccFormMessage> = () => ({
  components: { BccFormMessage },
  template: `
    <div class="inline-flex flex-col space-y-4">
      <BccFormMessage>
        <strong>Example message</strong> to show with an input
      </BccFormMessage>
      <BccFormMessage state="error">
        <strong>Example message</strong> to show with an input
      </BccFormMessage>
      <BccFormMessage state="success">
        <strong>Example message</strong> to show with an input
      </BccFormMessage>
      
    </div>
  `,
});
