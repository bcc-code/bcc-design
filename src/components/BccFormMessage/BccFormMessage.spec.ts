import { describe, it, expect } from "vitest";

import { mount } from "@vue/test-utils";
import BccFormMessage from "./BccFormMessage.vue";

describe("BccFormMessage", () => {
  it("renders a text from the default slot", () => {
    const wrapper = mount(BccFormMessage, {
      slots: { default: "Test message" },
    });

    expect(wrapper.text()).toBe("Test message");
    expect(wrapper.html()).toMatchSnapshot();
  });
});
