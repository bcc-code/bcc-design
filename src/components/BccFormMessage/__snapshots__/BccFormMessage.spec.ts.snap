// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BccFormMessage > renders a text from the default slot 1`] = `"<span class=\\"bcc-form-message bcc-form-message-default\\">Test message</span>"`;
